### DISCUSSION TITLE
How to inject form data to a modal view?

### DISCUSSION THREAD

**Discussion by NaungYeHtet** [filamentphp/filament#9368] [Help] [answered]
I have a filament form in a create page. Currently it has an action to open a modal with custom view.

**getFormActions function of CreatePage**

```
    protected function getFormActions(): array
    {
        return [
            Action::make('preview')
                ->icon('heroicon-o-eye')
                ->color('gray')
                ->modalContent(function (array $data): View {
                    dd($data);

                    return view(
                        'filament.pages.actions.preview',
                        ['data' => $data],
                    );
                }),

            $this->getCreateFormAction(),
            ...(static::canCreateAnother() ? [$this->getCreateAnotherFormAction()] : []),
            $this->getCancelFormAction(),
        ];
    }
```

Is there a way to inject data of form to the modal via action?

I tried using array $data to inject data but got empty array even though the form has data.

**Comment by sinnbeck** [comment]
Try using `$state` instead

**Reply by NaungYeHtet** [reply]
Hello @sinnbeck thanks you for answering. I got an error using $state.

`An attempt was made to evaluate a closure for [Filament\Actions\Action], but [$state] was unresolvable.`

I am using modal to preview form data before saving them to database.

So I need all form data to show the user if they are correct.

**Comment by NaungYeHtet** [comment, marked as answer]
I got the answer.
I am confusing with page actions with form actions

I have to declare an action inside `$form` instead of adding it into getFormActions.
So I got access to `$get` variable.

**Comment by photojo** [comment]
I've got a similar problem.

I call a modal 
`Action::make('config')
                        ->modalContent(fn (Action $action,Teacher $record): View => view(
                            'filament.pages.actions.advance',
                            ['action' => $action,'record' => $record],
                        ))`

and pass data to it. 
In the view I have
`@livewire('TeacherForm', ['id' => $record->id])`

It works fine, form is filled, but changes in the form are not saved. 
But they are saved, when I call the same form via route "directly".

How can I fix it?
